sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -498,6 +501,38 @@ private void SetNameFieldsIfNull()
             }
         }
 
+        #region internal static accessors for Azure.Storage.DataMovement.Blobs
+        /// <summary>
+        /// Get a <see cref="ShareFileClient"/>'s <see cref="HttpAuthorization"/>
+        /// for passing the authorization when performing service to service copy
+        /// where OAuth is necessary to authenticate the source.
+        /// </summary>
+        /// <param name="client">
+        /// The storage client which to generate the
+        /// authorization header off of.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The BlobServiceClient's HttpPipeline.</returns>
+        protected static async Task<HttpAuthorization> GetCopyAuthorizationHeaderAsync(
```

Comment:
Hm, I know I asked for this (cause I was under the impression that Files Service Copy supported OAuth but currently doesn't) but should we make this internal for now and then add it back later when it is supported?

It's just cause it's technically public API in the Files Package. So I would rather release it later when service copy is actually supported.

Reply:
well this is the only way anything works. If this goes we may as well just shelve the whole pr.